supervisor/misc/tasks.py

```diff
@@ -195,16 +189,30 @@ async def _watchdog_homeassistant_api(self):
             _LOGGER.warning("Watchdog missed an Home Assistant Core API response.")
             return
 
-        _LOGGER.error(
-            "Watchdog missed %s Home Assistant Core API responses in a row. Restarting Home Assistant Core API!",
-            HASS_WATCHDOG_MAX_API_ATTEMPTS,
-        )
+        if safe_mode := reanimate_fails == HASS_WATCHDOG_MAX_REANIMATE_ATTEMPTS:
+            _LOGGER.critical(
+                "Watchdog cannot reanimate Home Assistant Core, failed all %s attempts. Restarting into safe mode",
+                reanimate_fails,
+            )
+        else:
+            _LOGGER.error(
+                "Watchdog missed %s Home Assistant Core API responses in a row. Restarting Home Assistant Core API!",
```

Comment:
Was already before like this, but restarting the API sounds odd to me, no?

```suggestion
                "Watchdog missed %s Home Assistant Core API responses in a row. Restarting Home Assistant Core!",
```